src/Multicall3.sol

```diff
@@ -78,34 +111,71 @@ contract Multicall3 {
         }
     }
 
+    /// @notice Aggregate calls with a msg value
+    /// @notice Reverts if msg.value is less than the sum of the call values
+    /// @param calls An array of Call3Value structs
+    /// @return returnData An array of Result structs
+    function aggregate3Value(Call3Value[] calldata calls) public payable returns (Result[] memory returnData) {
+        uint256 valAccumulator;
+        returnData = new Result[](calls.length);
+        uint256 length = calls.length;
+        Call3Value calldata call;
+        for (uint256 i = 0; i < length;) {
+            Result memory result = returnData[i];
+            call = calls[i];
+            valAccumulator += call.value;
+            require(msg.value >= valAccumulator, "Multicall3: aggregate3Value failed");
```

Comment:
Should this be `==` check after the for loop instead of `>=` in the loop? Trying to think of a use case where you wouldn't want `==` 🤔

Reply:
Wait no, if there are multiple calls, then the total aggregate value (msg.value) should equal the sum of all the individual `call[i].value`s right?

This is basically checking: is the sum of `call[0...i].value` less than or equal to the `msg.value`

Reply:
The current form of this check allows `msg.value` to be greater than sum of `call[0...i].value`, but I'm not sure there is ever a use case for that? In other words, a stricter check that `msg.value` is exactly equal to the sum of `call[0...i].value` seems to provide safer guarantees (e.g. no ETH sent not allocated to a call), and probably saves gas since it's less total checks (1 check, instead of 1 each loop iteration)